playground/infrastructure/datastore_client.py

```diff
@@ -83,54 +87,67 @@ def save_to_cloud_datastore(self, examples_from_rep: List[Example], sdk: Sdk, or
         examples_ids_before_updating = self._get_all_examples(sdk, origin)
 
         # loop through every example to save them to the Cloud Datastore
-        with self._datastore_client.transaction():
-            for example in tqdm(examples_from_rep):
-                sdk_key = self._get_key(DatastoreProps.SDK_KIND, Sdk.Name(example.sdk))
-                example_id = self._make_example_id(origin, sdk, example.name)
-                updated_example_ids.append(example_id)
-                examples.append(
-                    self._to_example_entity(example, example_id, sdk_key, actual_schema_version_key, origin)
-                )
-                snippets.append(
-                    self._to_snippet_entity(example, example_id, sdk_key, now, actual_schema_version_key, origin)
-                )
-                pc_objects.extend(self._pc_object_entities(example, example_id))
-                files.append(self._to_file_entity(example, example_id))
-
-            self._datastore_client.put_multi(examples)
-            self._datastore_client.put_multi(snippets)
-            self._datastore_client.put_multi(pc_objects)
-            self._datastore_client.put_multi(files)
-
-            # delete examples from the Cloud Datastore that are not in the repository
-            examples_ids_for_removing = list(filter(lambda key: key not in updated_example_ids, examples_ids_before_updating))
-            if len(examples_ids_for_removing) != 0:
-                logging.info("Start of deleting extra playground examples ...")
-                examples_keys_for_removing = list(
-                    map(lambda ex_id: self._get_key(DatastoreProps.EXAMPLE_KIND, ex_id), examples_ids_for_removing)
-                )
-                snippets_keys_for_removing = list(
-                    map(lambda ex_id: self._get_key(DatastoreProps.SNIPPET_KIND, ex_id), examples_ids_for_removing)
-                )
-                file_keys_for_removing = list(
-                    map(self._get_files_key, examples_ids_for_removing)
+
+        for example in tqdm(examples_from_rep):
+            sdk_key = self._get_key(DatastoreProps.SDK_KIND, Sdk.Name(example.sdk))
+            example_id = self._make_example_id(origin, sdk, example.name)
+            updated_example_ids.append(example_id)
+            examples.append(
+                self._to_example_entity(example, example_id, sdk_key, actual_schema_version_key, origin)
+            )
+            snippet = self._to_snippet_entity(example, example_id, sdk_key, now, actual_schema_version_key, origin)
+            pc_objects.extend(self._pc_object_entities(example, example_id))
+            files.append(self._to_file_entity(example, example_id))
+            if example.datasets and example.emulators:
```

Comment:
`save_to_cloud_datastore` method saves both in Datastore and in Cloud Storage :(